src/gui/uithememanager.cpp

```diff
@@ -99,29 +123,30 @@ QIcon UIThemeManager::getIcon(const QString &iconId, const QString &fallback) co
 {
 #if (defined(Q_OS_UNIX) && !defined(Q_OS_MAC))
     if (m_useSystemTheme) {
-        QIcon icon = QIcon::fromTheme(iconId);
+        QIcon icon = QIcon::fromTheme(m_iconMap[iconId].toString());
         if (icon.name() != iconId)
-            icon = QIcon::fromTheme(fallback, QIcon(IconProvider::instance()->getIconPath(iconId)));
+            icon = QIcon::fromTheme(fallback, QIcon(getIconPath(iconId)));
```

Comment:
Also how do you suppose to provide fallback icon now? It seems that it should come from (system theme) icon mapping now.

Reply:
Should I add separate icon id for fallback as well? I think there are only like 4 instances where fallback is actually used.

Well, It does make sense to add IconIds for these as well. Will replace direct references with Icon ids.